cmd/util/ledger/migrations/cadence_values_migration.go

```diff
@@ -0,0 +1,298 @@
+	oldPayloads []*ledger.Payload,
+) ([]*ledger.Payload, error) {
+
+	// Create all the runtime components we need for the migration
+	migrationRuntime, err := newMigratorRuntime(address, oldPayloads)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create migrator runtime: %w", err)
+	}
+
+	migration := migrations.NewStorageMigration(
+		migrationRuntime.Interpreter,
+		migrationRuntime.Storage,
+	)
+
+	reporter := newValueMigrationReporter(m.reporter, m.log)
+
+	m.log.Info().Msg("Migrating cadence values")
+
+	migration.Migrate(
+		&migrations.AddressSliceIterator{
+			Addresses: []common.Address{
+				address,
+			},
+		},
+		migration.NewValueMigrationsPathMigrator(
+			reporter,
+			m.valueMigrations(migrationRuntime.Interpreter, migrationRuntime.Accounts, reporter)...,
+		),
+	)
+
+	m.log.Info().Msg("Committing changes")
+	err = migration.Commit()
+	if err != nil {
+		return nil, fmt.Errorf("failed to commit changes: %w", err)
+	}
+
+	// finalize the transaction
+	result, err := migrationRuntime.TransactionState.FinalizeMainTransaction()
+	if err != nil {
+		return nil, fmt.Errorf("failed to finalize main transaction: %w", err)
+	}
+
+	// Merge the changes to the original payloads.
+	return m.mergeRegisterChanges(migrationRuntime, result.WriteSet)
+}
+
+func (m *CadenceBaseMigrator) mergeRegisterChanges(
+	mr *migratorRuntime,
+	changes map[flow.RegisterID]flow.RegisterValue,
+) ([]*ledger.Payload, error) {
+
+	originalPayloads := mr.Snapshot.Payloads
+	newPayloads := make([]*ledger.Payload, 0, len(originalPayloads))
+
+	// Add all new payloads.
+	for id, value := range changes {
+		key := convert.RegisterIDToLedgerKey(id)
+		newPayloads = append(newPayloads, ledger.NewPayload(key, value))
+	}
+
+	// Add any old payload that wasn't updated.
+	for id, value := range originalPayloads {
+		if len(value.Value()) == 0 {
+			// This is strange, but we don't want to add empty values. Log it.
+			m.log.Warn().Msgf("empty value for key %s", id)
+			continue
+		}
+
+		// If the payload had changed, then it has been added earlier.
+		// So skip old payload.
+		if _, contains := changes[id]; contains {
+			continue
+		}
+
+		newPayloads = append(newPayloads, value)
+	}
+
+	return newPayloads, nil
+}
+
+func NewCadenceValueMigrator(
+	rwf reporters.ReportWriterFactory,
+	capabilityIDs map[interpreter.AddressPath]interpreter.UInt64Value,
+) *CadenceBaseMigrator {
+	return &CadenceBaseMigrator{
+		reporter: rwf.ReportWriter("cadence-value-migrator"),
+		valueMigrations: func(
+			inter *interpreter.Interpreter,
+			_ environment.Accounts,
+			reporter *cadenceValueMigrationReporter,
+		) []migrations.ValueMigration {
+			// All cadence migrations except the `capcons.LinkValueMigration`.
+			return []migrations.ValueMigration{
+				&capcons.CapabilityValueMigration{
+					CapabilityIDs: capabilityIDs,
+					Reporter:      reporter,
+				},
+				// Must be run before account-type migration
+				entitlements.NewEntitlementsMigration(inter),
+
```

Comment:
Order now shouldn't mater anymore, right?

```suggestion
				entitlements.NewEntitlementsMigration(inter),
```